packages/client/hmi-client/src/components/code/tera-code.vue

```diff
@@ -158,26 +209,39 @@ const emit = defineEmits(['asset-loaded']);
 
 const toast = useToastService();
 
+const existingMarkers = new Set();
+
 const codeName = ref('');
 const codeText = ref(INITIAL_TEXT);
 const codeAsset = ref<Code | null>(null);
 const editor = ref<VAceEditorInstance['_editor'] | null>(null);
 const selectedText = ref('');
+const selectionRange = ref<Ace.Range | null>(null);
 const progress = ref(0);
 const isCodeToModelLoading = ref(false);
+const willGenerateFromDynamics = ref(false);
 const isModelDiagramModalVisible = ref(false);
 const isModelNamingModalVisible = ref(false);
 const isCodeNamingModalVisible = ref(false);
+const isDynamicsModalVisible = ref(false);
 const newCodeName = ref('');
 const newModelName = ref('');
 const newModelDescription = ref('');
+const newDynamicsName = ref('');
+const newDynamicsDescription = ref('');
 const programmingLanguage = ref<ProgrammingLanguage>(ProgrammingLanguage.Python);
 const programmingLanguages = [
 	ProgrammingLanguage.Julia,
 	ProgrammingLanguage.Python,
 	ProgrammingLanguage.R
 ];
 
+const selectedRangeToString = computed(() =>
+	selectionRange.value
+		? `L${selectionRange.value.start.row + 1}-L${selectionRange.value.end.row + 1}`
+		: ''
+);
```

Comment:
@Sorrento110 I remember you mentioning that we are moving away from GitHub style of line selection.
Could you point us to the new way to write those please?